### PR TITLE
Add contribution guide

### PR DESCRIPTION
**Pull request by davesnx**
While doing https://github.com/reasonml-labs/decco/pull/64 I annotate some of the steps to get it working.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+- [Node](https://nodejs.org/en) and [yarn](https://yarnpkg.com)
+- [esy](https://esy.sh)
+
+### Developing
+
+- git clone https://github.com/reasonml-labs/decco
+- cd decco
+- yarn install
+- cd ppx_src; esy;
```
Comment (ryb73) [inline CONTRIBUTING.md:23]:
```suggestion
- yarn build-ppx
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+`ocaml-lsp-server` is only needed for development.
+
+You would need to have [opam](https://opam.ocaml.org) installed.
+
+- cd ppx_src;
+- opam switch create . 4.06.0 --deps-only # If it's the first time you create a switch from 4.06, it can take a while.
+- eval $(opam env)
+- opam install -y ocaml-lsp-server dune ocaml ppx_tools_versioned reason # Install ppx's dependencies inside the switch
+- Profit, this should make your editor a little more smart
```
Comment (ryb73) [inline CONTRIBUTING.md:43]: Is this all that needs to be done? No need to install an editor extension or anything? I've never used ocaml-lsp

File: README.md [hunk 3]
```diff
@@ -127,3 +127,7 @@ When decoding a record, the default value will be used for keys that are missing
 let {s} = Js.Json.parseExn("{}") |> record_decode |> Belt.Result.getExn;
 Js.log(s); /* def */
 ```
+
+# Contributing
+
+Please read the [CONTRIBUTE]
```
Comment (ryb73) [inline README.md:133]: broken link?

**Review by ryb73 — COMMENTED** [review]
This is great! Left a few comments

**Comment by ryb73** [comment]
Thanks!

**Comment by davesnx** [comment]
Thanks to you, after this PR got merged https://github.com/esy-ocaml/esy-opam-override/pull/88 is now possible to install ocaml-lsp with esy. I might open another PR to remove the opam instructions!

**Pull request merged**